src/Logging/EntityListener/HearingPostListener.php

```diff
@@ -15,12 +18,32 @@ public function __construct(Security $security)
 
     public function postPersist(HearingPost $hearingPost, LifecycleEventArgs $args)
     {
-        $this->logActivity('Hearing post created', $args);
+        $action = sprintf('%s created', $this->getHearingPostType($hearingPost));
+        $this->logActivity($action, $args);
     }
 
     public function postUpdate(HearingPost $hearingPost, LifecycleEventArgs $args)
     {
-        $this->logActivity('Hearing post updated', $args);
+        $action = sprintf('%s updated', $this->getHearingPostType($hearingPost));
+        $this->logActivity($action, $args);
+    }
+
+    public function preRemove(HearingPost $hearingPost, LifecycleEventArgs $args)
+    {
+        $action = sprintf('%s deleted', $this->getHearingPostType($hearingPost));
+        $this->logActivity($action, $args);
+    }
+
+    private function getHearingPostType(HearingPost $hearingPost)
+    {
+        if ($hearingPost instanceof HearingPostRequest) {
+            return 'Hearing post request';
+        } elseif ($hearingPost instanceof HearingPostResponse) {
+            return 'Hearing post response';
+        }
+
+        $message = sprintf('Unhandled hearing post type %s found.', get_class($hearingPost));
+        throw new ItkDevLoggingException($message);
```

Comment:
Why don't you use your cool `return match(true) { … }` trick here?

Reply:
Used cool trick 😎